op-challenger/challenger/output_test.go

```diff
@@ -10,13 +10,49 @@ import (
 	"github.com/stretchr/testify/mock"
 	"github.com/stretchr/testify/require"
 
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
 	"github.com/ethereum/go-ethereum/log"
 
+	"github.com/ethereum-optimism/optimism/op-bindings/bindings"
 	"github.com/ethereum-optimism/optimism/op-challenger/metrics"
 	"github.com/ethereum-optimism/optimism/op-node/eth"
 	"github.com/ethereum-optimism/optimism/op-node/testlog"
 )
 
+func TestParseOutputLog_Succeeds(t *testing.T) {
+	challenger := newTestChallenger(t, eth.OutputResponse{}, true)
+	expectedBlockNumber := big.NewInt(0x04)
+	expectedOutputRoot := eth.Bytes32{0x02}
+	logTopic := challenger.l2ooABI.Events["OutputProposed"].ID
+	log := types.Log{
+		Topics: []common.Hash{logTopic, common.Hash(expectedOutputRoot), {0x03}, common.BigToHash(expectedBlockNumber)},
+	}
+	parsedBlockNumber, parsedOutputRoot, err := challenger.ParseOutputLog(&log)
+	require.NoError(t, err)
+	require.Equal(t, expectedBlockNumber, parsedBlockNumber)
+	require.Equal(t, expectedOutputRoot, parsedOutputRoot)
+}
+
+func TestParseOutputLog_WrongLogTopic_Errors(t *testing.T) {
+	challenger := newTestChallenger(t, eth.OutputResponse{}, true)
+	_, _, err := challenger.ParseOutputLog(&types.Log{
+		Topics: []common.Hash{{0x01}, {0x02}, {0x03}, {0x04}},
+	})
+	require.ErrorIs(t, err, ErrInvalidOutputLogTopic)
+}
+
+func TestParseOutputLog_BadLog_Panics(t *testing.T) {
```

Comment:
We should not panic in parsing code, please return an error instead. Note that we don't always need named errors unless we expect the end user of this API to inspect the retuned error & do something based on the specific (rather than presence) error value.